firebase-installations/build.gradle.kts

```diff
@@ -0,0 +1,162 @@
+    get() = if (this is KonanTarget.IOS_X64 || this is KonanTarget.IOS_SIMULATOR_ARM64) {
+        "ios-arm64_i386_x86_64-simulator"
+    } else {
+        "ios-arm64_armv7"
+    }
+
+kotlin {
+
+    android {
+        publishAllLibraryVariants()
+    }
+
+    val supportIosTarget = project.property("skipIosTarget") != "true"
+    if (supportIosTarget) {
+        fun nativeTargetConfig(): KotlinNativeTarget.() -> Unit = {
+            val nativeFrameworkPaths = listOf(
+                rootProject.project("firebase-app").projectDir.resolve("src/nativeInterop/cinterop/Carthage/Build/iOS")
+            ).plus(
+                listOf(
+                    "FirebaseAnalytics",
+                    "FirebaseCore",
+                    "FirebaseCoreDiagnostics",
+                    "FirebaseInstallations",
+                    "GoogleAppMeasurement",
+                    "GoogleAppMeasurementIdentitySupport",
+                    "GoogleDataTransport",
+                    "GoogleUtilities",
+                    "nanopb",
+                    "PromisesObjC"
+                ).map {
+                    rootProject.project("firebase-app").projectDir.resolve("src/nativeInterop/cinterop/Carthage/Build/$it.xcframework/${konanTarget.archVariant}")
+                }
+            )
+
+            binaries {
+                getTest("DEBUG").apply {
+                    linkerOpts(nativeFrameworkPaths.map { "-F$it" })
+                    linkerOpts("-ObjC")
+                }
+            }
+
+            compilations.getByName("main") {
+                cinterops.create("FirebaseInstallations") {
+                    compilerOpts(nativeFrameworkPaths.map { "-F$it" })
+                    extraOpts = listOf("-compiler-option", "-DNS_FORMAT_ARGUMENT(A)=", "-verbose")
+                }
+            }
+        }
+
+        ios(configure = nativeTargetConfig())
+        iosSimulatorArm64(configure = nativeTargetConfig())
+    }
+
+    js {
+        useCommonJs()
+        nodejs {
+            testTask {
+                useMocha {
+                    timeout = "5s"
+                }
+            }
+        }
+        browser {
+            testTask {
+                useMocha {
+                    timeout = "5s"
+                }
+            }
+        }
+    }
+
+    sourceSets {
+        all {
+            languageSettings.apply {
+                apiVersion = "1.5"
+                languageVersion = "1.5"
+                progressiveMode = true
+//                optIn("kotlinx.coroutines.ExperimentalCoroutinesApi")
+//                optIn("kotlinx.serialization.InternalSerializationApi")
+            }
+        }
+
+        val commonMain by getting {
+            dependencies {
+                api(project(":firebase-app"))
+                implementation(project(":firebase-common"))
+            }
+        }
+
+        val androidMain by getting {
+            dependencies {
+                api("com.google.firebase:firebase-installations")
```

Comment:
@nbransby Actually there's a separate package for Android, so probably makes sense to leave it as separate module.

Reply:
@nbransby Can you take a look? I have some more PRs in my pipeline :)